### PR TITLE
Updates to route53 resource for enabling DNS Query logging

### PR DESCRIPTION
**Pull request by eisjcormier**
Added a filter determining DNS Query logging configuration state and action for enabling  query logging.  
Added unit test and placebo data as well

### CODE REVIEW

**Comment by cliffpracht** [comment]
Nice work @eisjcormier!   This looks good to me.  I think the ultimate test will be what Kapil thinks.  Did you create an issue/PR to main?

File: c7n/resources/route53.py [hunk 1]
```diff
@@ -95,6 +97,93 @@ def get_arns(self, resource_set):
             arns.append(self.generate_arn(_id))
         return arns
 
+@HostedZone.action_registry.register('enable-query-logging')
+class EnableQueryLogging(BaseAction):
+
+    permissions = ('route53:GetQueryLoggingConfig','route53:GetHostedZone','route53:CreateQueryLoggingConfig',
+        'route53:DeleteQueryLoggingConfig','logs:CreateLogGroup','logs:DescribeLogGroups','logs:PutRetentionPolicy')
+
+    schema = type_schema('enable-query-logging', state={'type': 'boolean'}, logretentiondays={'type': 'number'})
```
Comment (dfengshi) [inline c7n/resources/route53.py:106]: Any thoughts on using the hyphen naming scheme here for logretentiondays in the type_schema?

**Comment by dfengshi** [comment]
@eisjcormier I spent some time looking through the logic and tests, and this looks good to me. Great work! 

I have the same opinion as Cliff - Kapil and co might have some valuable feedback, I think this looks good to submit to the main repo for review.

File: c7n/resources/route53.py (hunk 1 above)
Comment (eisjcormier) [reply c7n/resources/route53.py:106]: hyphens cause a "can't assign to operator" error.  Think I should use underscores?

Comment (dfengshi) [reply c7n/resources/route53.py:106]: I might be wrong, but I think it just needs to be 

```
schema = type_schema('enable-query-logging', state={'type': 'boolean'}, 'log-retention-days'={'type': 'number'})
```

Comment (eisjcormier) [reply c7n/resources/route53.py:106]: Hyphens will not work here though:
logretentiondays = self.data.get('log-retention-days', 30)

I was thinking to keep the var the same.

Comment (dfengshi) [reply c7n/resources/route53.py:106]: Ah I see, yeah let's keep it the same

**Pull request merged**